crates/legacy/src/block_size_limits.rs

```diff
@@ -0,0 +1,159 @@
+use atomic::Atomic;
```

Comment:
This was moved to its own file.  The only change is making it atomic to reduce amount of locks we juggle in the legacy builder.